sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -1271,6 +1271,10 @@ ConnectionPolicy buildConnectionPolicy() {
     private void validateConfig() {
         URI uri;
         try {
+            // split the serviceEndpoint in case of a connection string being passed in
+            String[] serviceEndpointParts = serviceEndpoint.replace(";", "")
+                .replace("AccountKey=", "").replace("https://","").split("/");
```

Comment:
Is there a specific reason we are replacing the "https://" and then again adding it?

Reply:
Went for the approach of only reconstructing when a path or query is specified to limit risk of break changes.